test/specs/wrapper/overview.spec.js

```diff
@@ -0,0 +1,336 @@
+import { describeWithShallowAndMount, vueVersion } from '~resources/utils'
+
+describeWithShallowAndMount('overview', mountingMethod => {
+  const originalLog = console.log
+  let consoleOutput = []
+  const consoleLogMock = (...output) =>
+    consoleOutput.push(...output.join(' ').split('\n'))
+
+  beforeEach(() => {
+    consoleOutput = []
+    console.log = consoleLogMock
+  })
+  afterEach(() => (console.log = originalLog))
+
+  it('throws error when called on non VueWrapper', () => {
+    const wrapper = mountingMethod({ template: '<div><p /></div>' })
+    const nonVueWrapper = wrapper.find('p')
+    const message =
+      '[vue-test-utils]: wrapper.overview() can only be called on a Vue instance'
+
+    expect(() => nonVueWrapper.overview())
+      .to.throw()
+      .with.property('message', message)
+  })
+
+  if (vueVersion > 2) {
```

Comment:
how come we need this check? This lib only supports Vue 2.x - doe s something weird happen if you remove this?

Reply:
The test of `test:compat` fails at Vue 2.0.
This is because hooks are added to the emitted:

```javascript
"Emitted: {"
	" hook:beforeCreate: ["
	" 0: [ ]"
	" ],"
	" hook:created: ["
	" 0: [ ]"
	" ],"
	" hook:beforeMount: ["
	" 0: [ ]"
	" ],"
	" hook:mounted: ["
	" 0: [ ]"
	" ]"
 "}"
```

I don't know exactly how many people use version 2.0. If we want to test it, I can think of the following options:

1. **Duplicate the tests (or a part of them) in an `else`**: We repeat a lot of code but gain in readability.

2.  **Extract the expectedConsoleOutput to a function (at least the emitted)**: The tests are less clear, but it's better than adding a condition to each test.

Reply:
I think I have a middle ground solution. 
Right now we're checking the outcome with this:

```javascript
expect(consoleOutput).to.have.ordered.members(expectedConsoleOutput)
```

If we replace the `have.ordered` with an `include`  it should work for Vue 2.0:
```
expect(consoleOutput).to.include.members(expectedConsoleOutput)
```

The only problem we have is when nothing has been emitted. In that case the expected emit will be `Emitted: {}` and in 2.0 it will show:

```javascript
"Emitted: {"
" hook:beforeCreate: ["
" 0: [ ]"
" ],"
" hook:created: ["
" 0: [ ]"
" ],"
" hook:beforeMount: ["
" 0: [ ]"
" ],"
" hook:mounted: ["
" 0: [ ]"
" ]"
"}"
```

My proposal is to create a simple function that takes care of this without making the tests dirty.

Something like:
```javascript
expect(compareConsoleOutput(consoleOutput, expected)).to.be.true
```
or even a function that includes the expect inside:
```javascript
expectToHaveConsoleOutputs(consoleOutput, expectedConsoleOutputs)
```

Reply:
I see - I think it's fine to leave the check in. I doubt many people are on 2.0 now anyway. No problem 👍